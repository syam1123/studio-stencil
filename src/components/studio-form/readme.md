# studio-form



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description             | Type      | Default     |
| ----------- | ------------ | ----------------------- | --------- | ----------- |
| `isLoading` | `is-loading` | Indicates loading state | `boolean` | `undefined` |


## Events

| Event           | Description | Type               |
| --------------- | ----------- | ------------------ |
| `formSubmitted` |             | `CustomEvent<any>` |


## Dependencies

### Depends on

- [studio-loader](../studio-loader)

### Graph
```mermaid
graph TD;
  studio-form --> studio-loader
  style studio-form fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
