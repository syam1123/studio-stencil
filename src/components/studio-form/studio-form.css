:host {
  display: block;
}

.container {
  display: flex;
  max-width: 1200px;
  width: 90%;
  margin: auto;
  justify-content: center;
}

.container form {
  display: grid;
  grid-gap: 1rem;
  margin: 2rem;
  border: 1px solid gray;
  background-color: #ccccff;
  padding: 2rem;
  border-radius: .4rem;
}