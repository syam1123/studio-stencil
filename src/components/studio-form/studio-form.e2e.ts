import { newE2EPage } from '@stencil/core/testing';

describe('studio-form', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<studio-form></studio-form>');

    const element = await page.find('studio-form');
    expect(element).toHaveClass('hydrated');
  });
});
