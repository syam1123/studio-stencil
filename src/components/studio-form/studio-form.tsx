import { Component, State, ComponentInterface, h, EventEmitter, Event, Prop } from '@stencil/core';

@Component({
  tag: 'studio-form',
  styleUrl: 'studio-form.css',
  shadow: true,
})

export class StudioForm implements ComponentInterface {
  /**
   * Indicates loading state
   */
  @Prop() isLoading: boolean;
  selectedReceiverIds = [102, 103];
  @State() userObj: Object = {
    firstName: {
      label: 'First Name',
      type: 'text',
      value: ''
    },
    lastName: {
      label: 'Last Name',
      type: 'text',
      value: ''
    },
    email: {
      label: 'Email',
      type: 'email',
      value: ''
    },
    phoneNumber: {
      label: 'Phone Number',
      type: 'text',
      value: ''
    },
  };
  @Event({
    eventName: 'formSubmitted',
    composed: true,
    cancelable: true,
    bubbles: true,
  }) formSubmitted: EventEmitter<any>;

  handleSubmit(e) {
    e.preventDefault();
    this.formSubmitted.emit({
      user: this.userObj
    })
  }

  handleChange(event, key) {
    this.userObj[key] = event.target.value;
  }

  render() {
    if (this.isLoading) {
      return (
        <div class="container">
          <studio-loader />
        </div>
      )
    }
    return (
      <div class="container">
        <form onSubmit={(e) => this.handleSubmit(e)}>
          {Object.keys(this.userObj).map((field) => (
            <label>
              {this.userObj[field].label}:
              <input type={this.userObj[field].type} value={this.userObj[field].value} onInput={(e) => this.handleChange(e, field)} />
            </label>
          ))}
          <input type="submit" value="Submit" />
        </form>
      </div>
    );
  }

}
