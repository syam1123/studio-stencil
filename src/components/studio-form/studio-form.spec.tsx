import { newSpecPage } from '@stencil/core/testing';
import { StudioForm } from './studio-form';

describe('studio-form', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [StudioForm],
      html: `<studio-form></studio-form>`,
    });
    expect(page.root).toMatchSnapshot()
  });
});
