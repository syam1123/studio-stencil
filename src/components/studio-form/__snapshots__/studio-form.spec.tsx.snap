// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`studio-form renders 1`] = `
<studio-form>
  <mock:shadow-root>
    <div class="container">
      <form>
        <label>
          First Name:
          <input type="text" value="">
        </label>
        <label>
          Last Name:
          <input type="text" value="">
        </label>
        <label>
          Email:
          <input type="email" value="">
        </label>
        <label>
          Phone Number:
          <input type="text" value="">
        </label>
        <input type="submit" value="Submit">
      </form>
    </div>
  </mock:shadow-root>
</studio-form>
`;
